tests/aws/services/cloudformation/api/test_update_stack.py

```diff
@@ -407,3 +408,43 @@ def test_update_with_rollback_configuration(deploy_cfn_template, aws_client):
 
     # cleanup
     aws_client.cloudwatch.delete_alarms(AlarmNames=["HighResourceUsage"])
+
+
+@markers.aws.validated
+def test_diff_after_update(deploy_cfn_template, aws_client, snapshot):
+    template_1 = textwrap.dedent("""
+    Resources:
+        SimpleParam:
+            Type: AWS::SSM::Parameter
+            Properties:
+                Value: test
+                Type: String
+    """)
+    template_2 = textwrap.dedent("""
+    Resources:
+        SimpleParam1:
+            Type: AWS::SSM::Parameter
+            Properties:
+                Value: test1
+                Type: String
+    """)
+
+    stack = deploy_cfn_template(
+        template=template_1,
+    )
+    aws_client.cloudformation.get_waiter("stack_create_complete").wait(StackName=stack.stack_name)
+    aws_client.cloudformation.update_stack(
+        StackName=stack.stack_name,
+        TemplateBody=template_2,
+    )
+    aws_client.cloudformation.get_waiter("stack_update_complete").wait(StackName=stack.stack_name)
+    get_template_response_1 = aws_client.cloudformation.get_template(StackName=stack.stack_name)
+    snapshot.match("get-template-response-1", get_template_response_1)
+
+    with pytest.raises(botocore.exceptions.ClientError):
+        aws_client.cloudformation.update_stack(
+            StackName=stack.stack_name,
+            TemplateBody=template_2,
+        )
```

Comment:
It would be good to capture the error response, e.g.

```suggestion
    with pytest.raises(botocore.exceptions.ClientError) as exc_info:
        aws_client.cloudformation.update_stack(
            StackName=stack.stack_name,
            TemplateBody=template_2,
        )
    snapshot.match("update-2-error", exc_info.value.response)
```

Reply:
Error from localstack was not in parity with AWS. I've updated the parity issue.